providers/common/ciphers/cipher_aes_xts.c

```diff
@@ -0,0 +1,286 @@
+static OSSL_OP_cipher_get_params_fn aes_##kbits##_##lcmode##_get_params;       \
+static int aes_##kbits##_##lcmode##_get_params(OSSL_PARAM params[])            \
+{                                                                              \
+    return cipher_generic_get_params(params, EVP_CIPH_##UCMODE##_MODE,         \
+                                     flags, 2 * kbits, AES_XTS_BLOCK_BITS,     \
+                                     AES_XTS_IV_BITS);                         \
+}                                                                              \
+static OSSL_OP_cipher_newctx_fn aes_##kbits##_xts_newctx;                      \
+static void *aes_##kbits##_xts_newctx(void *provctx)                           \
+{                                                                              \
+    return aes_xts_newctx(provctx, EVP_CIPH_##UCMODE##_MODE, 2 * kbits,        \
+                          AES_XTS_BLOCK_BITS, AES_XTS_IV_BITS);                \
+}                                                                              \
+const OSSL_DISPATCH aes##kbits##xts_functions[] = {                            \
+    { OSSL_FUNC_CIPHER_NEWCTX, (void (*)(void))aes_##kbits##_xts_newctx },     \
+    { OSSL_FUNC_CIPHER_ENCRYPT_INIT, (void (*)(void))aes_xts_einit },          \
+    { OSSL_FUNC_CIPHER_DECRYPT_INIT, (void (*)(void))aes_xts_dinit },          \
+    { OSSL_FUNC_CIPHER_UPDATE, (void (*)(void))aes_xts_stream_update },        \
+    { OSSL_FUNC_CIPHER_FINAL, (void (*)(void))aes_xts_stream_final },          \
+    { OSSL_FUNC_CIPHER_CIPHER, (void (*)(void))aes_xts_cipher },               \
+    { OSSL_FUNC_CIPHER_FREECTX, (void (*)(void))aes_xts_freectx },             \
+    { OSSL_FUNC_CIPHER_DUPCTX, (void (*)(void))aes_xts_dupctx },               \
+    { OSSL_FUNC_CIPHER_GET_PARAMS,                                             \
+      (void (*)(void))aes_##kbits##_##lcmode##_get_params },                   \
+    { OSSL_FUNC_CIPHER_GETTABLE_PARAMS,                                        \
+      (void (*)(void))cipher_generic_gettable_params },                        \
+    { OSSL_FUNC_CIPHER_GET_CTX_PARAMS,                                         \
+      (void (*)(void))cipher_generic_get_ctx_params },                         \
+    { OSSL_FUNC_CIPHER_GETTABLE_CTX_PARAMS,                                    \
+      (void (*)(void))cipher_generic_gettable_ctx_params },                    \
+    { OSSL_FUNC_CIPHER_SET_CTX_PARAMS,                                         \
+      (void (*)(void))aes_xts_set_ctx_params },                                \
+    { OSSL_FUNC_CIPHER_SETTABLE_CTX_PARAMS,                                    \
+     (void (*)(void))aes_xts_settable_ctx_params },                            \
+    { 0, NULL }                                                                \
+}
+
+IMPLEMENT_cipher(xts, XTS, 256, AES_XTS_FLAGS);
+IMPLEMENT_cipher(xts, XTS, 128, AES_XTS_FLAGS);
```

Comment:
I wonder, is XTS an AES specific mode? If not, you (well, someone really, but since you currently hold the title "the man that..." :wink:) will need to refractor at some point. Question is if you want to do that now or later...

Reply:
Refactor later if needed - I have only seen it used by AES.

Reply:
Okie.